lib/active_merchant/billing/gateways/worldpay_online_payments.rb

```diff
@@ -141,7 +141,23 @@ def commit(method, url, parameters=nil, options = {})
 
           if (raw_response != '')
             response = parse(raw_response)
-            success = !response.key?("httpStatusCode")
+            if type=='token'
+              success = response.key?("token")
+            elsif
+              if response.key?("httpStatusCode")
+                success = false
+              elsif
+                if type=='authorize'
+                  success = response["paymentStatus"]=="AUTHORIZED" ? true: false
+                elsif type=='purchase'
+                  success = response["paymentStatus"]=="SUCCESS" ? true: false
+                elsif type=='capture' || type=='refund' || type=='void'
+                  success = true
```

Comment:
They don't contain a response body, merely a 200 and skip the httpStatusCode check above